ush/python/pygw/src/pygw/jinja.py

```diff
@@ -1,44 +1,191 @@
+        render_map = {'stream': self._render_stream,
+                      'file': self._render_file}
+        return render_map[self.template_type]()
+
+    def get_set_env(self, loader: jinja2.BaseLoader) -> jinja2.Environment:
+        """
+        Description
+        -----------
+        Define the environment for the jinja2 template
+        Any number of filters can be added here
+
+        Parameters
+        ----------
+        loader: of class jinja2.BaseLoader
+        Returns
+        -------
+        env: jinja2.Environment
+        """
+        env = jinja2.Environment(loader=loader, undefined=self.undefined)
+        env.filters["strftime"] = lambda dt, fmt: strftime(dt, fmt)
+        env.filters["to_isotime"] = lambda dt: to_isotime(dt) if not isinstance(dt, SilentUndefined) else dt
+        env.filters["to_fv3time"] = lambda dt: to_fv3time(dt) if not isinstance(dt, SilentUndefined) else dt
+        env.filters["to_YMDH"] = lambda dt: to_YMDH(dt) if not isinstance(dt, SilentUndefined) else dt
+        env.filters["to_YMD"] = lambda dt: to_YMD(dt) if not isinstance(dt, SilentUndefined) else dt
+        return env
+
+    @staticmethod
+    def add_filter_env(env: jinja2.Environment, filter_name: str, filter_func: callable):
+        """
+        Description
+        -----------
+        Add a custom filter to the jinja2 environment
+        Not implemented yet.  Placed here for future use
+        Parameters
+        ----------
+        env: jinja2.Environment
+            Active jinja2 environment
+        filter_name: str
+            name of the filter
+        filter_func: callable
+            function that will be called
+        Returns
+        -------
+        env: jinja2.Environment
+            Active jinja2 environment with the new filter added
+        """
+        raise NotImplementedError("Not implemented yet.  Placed here for future use")
+        # Implementation would look something like the following
+        # env.filters[filter_name] = filter_func
+        # return env
 
     def _render_stream(self):
-        raise NotImplementedError("Unable to handle templates other than files")
+        loader = jinja2.BaseLoader()
+        env = self.get_set_env(loader)
+        template = env.from_string(self.template_stream)
+        return self._render_template(template)
 
-    def _render_file(self):
+    def _render_file(self, data: dict = None):
         template_dir = self.template_path.parent
         template_file = self.template_path.relative_to(template_dir)
 
-        dirname = os.path.dirname(str(self.template_path))
-        relpath = os.path.relpath(str(self.template_path), dirname)
-
         loader = jinja2.FileSystemLoader(template_dir)
-        output = self._render(str(template_file), loader)
+        env = self.get_set_env(loader)
+        template = env.get_template(str(template_file))
+        return self._render_template(template)
 
-        return output
+    def _render_template(self, template):
```

Comment:
`def _render_template(self, template):` should be `def _render_template(self, template: jinja2.Template):`